server/data/data-cassandra/src/main/java/org/apache/james/sieve/cassandra/CassandraSieveDAO.java

```diff
@@ -0,0 +1,293 @@
+        return CassandraUtils.executeAsyncFirstRow(session,
+            selectActiveScriptMetadataStatement.bind()
+                .setString(CassandraSieveTable.USER_NAME, user)
+                .setBool(CassandraSieveTable.IS_ACTIVE, true));
+    }
+
+    public CompletableFuture<Boolean> deleteScriptInCassandra(String user, String name) {
+        return CassandraUtils.executeAsyncReturnApplied(session,
+            deleteScriptStatement.bind()
+                .setString(CassandraSieveTable.USER_NAME, user)
+                .setString(CassandraSieveTable.SCRIPT_NAME, name));
+    }
+
+    public CompletableFuture<Boolean> scriptExists(String user, String name) {
+        return getScriptMetadataRow(user, name).thenApply(Optional::isPresent);
+    }
+
+    public CompletableFuture<Optional<Long>> getQuota() {
+        return CassandraUtils.executeAsyncFirstRow(session,
+            selectClusterQuotaStatement.bind()
+                .setString(CassandraSieveClusterQuotaTable.NAME, CassandraSieveClusterQuotaTable.DEFAULT_NAME))
+            .thenApply(optional -> optional.map(row -> row.getLong(CassandraSieveClusterQuotaTable.VALUE)));
+    }
+
+    public CompletableFuture<Void> setQuota(long quota) {
+        return CassandraUtils.executeVoidAsync(session,
+            updateClusterQuotaStatement.bind()
+                .setLong(CassandraSieveClusterQuotaTable.VALUE, quota)
+                .setString(CassandraSieveClusterQuotaTable.NAME, CassandraSieveClusterQuotaTable.DEFAULT_NAME));
+    }
+
+    public CompletableFuture<ResultSet> removeQuota() {
+        return CassandraUtils.executeAsync(session,
+            deleteClusterQuotaStatement.bind()
+                .setString(CassandraSieveClusterQuotaTable.NAME, CassandraSieveClusterQuotaTable.DEFAULT_NAME));
+    }
+
+    public CompletableFuture<Optional<Long>> getQuota(String user) {
+        return CassandraUtils.executeAsyncFirstRow(session,
+            selectUserQuotaStatement.bind()
+                .setString(CassandraSieveQuotaTable.USER_NAME, user))
+            .thenApply(optional -> optional.map(row -> row.getLong(CassandraSieveQuotaTable.QUOTA)));
+    }
+
+    public CompletableFuture<Optional<String>> getActiveName(String user) {
+        return CassandraUtils.executeAsyncFirstRow(session,
+            selectActiveScriptNameStatement.bind()
+                .setString(CassandraSieveTable.USER_NAME, user)
+                .setBool(CassandraSieveTable.IS_ACTIVE, true))
+            .thenApply(optional -> optional.map(row -> row.getString(CassandraSieveTable.SCRIPT_NAME)));
+    }
+
+    public CompletableFuture<Void> setQuota(String user, long quota) {
+        return CassandraUtils.executeVoidAsync(session,
+            updateUserQuotaStatement.bind()
+                .setLong(CassandraSieveQuotaTable.QUOTA, quota)
+                .setString(CassandraSieveQuotaTable.USER_NAME, user));
+    }
+
+    public CompletableFuture<ResultSet> removeQuota(String user)  {
```

Comment:
how the caller is supposed to know what fields to expect from the rows in the resultset ?